lib/livebook_web/channels/live_socket.ex

```diff
@@ -0,0 +1,23 @@
+defmodule LivebookWeb.LiveSocket do
+  use Phoenix.Socket
+
+  # LiveView channels
+  channel "lvu:*", Phoenix.LiveView.UploadChannel
+  channel "lv:*", Phoenix.LiveView.Channel
+
+  @impl true
+  def connect(params, socket, info) do
+    auth_mode = Livebook.Config.auth_mode()
+
+    if LivebookWeb.AuthPlug.authenticated?(info.session || %{}, info.uri.port, auth_mode) do
+      Phoenix.LiveView.Socket.connect(params, socket, info)
+    else
+      :error
+    end
+  end
+
+  @impl true
+  def id(socket) do
+    Phoenix.LiveView.Socket.id(socket)
```

Comment:
I think we can kill this module altogether now and rollback to using Phoenix.LiveView.Socket. The authentication we already do in the LiveView itself, don't we?

Reply:
We currently authenticate only here (we have the UserHook, but it only builds the user data, no auth there). We added it here as it covered both use cases. With our auth model checking on socket level works and this ensures we don't miss that on any LV. wdyt?

Reply:
I see, so now that we have to duplicate the logic, I would remove this one then because what we are doing here is not public API. :D So move this to `UserHook` or `on_mount` callback and get rid of this particular socket. :)